packages/build-utils/src/detect-file-system-api.ts

```diff
@@ -168,24 +169,44 @@ export async function detectFileSystemAPI({
         reason: `Detected Next.js with Output Directory \`${projectSettings.outputDirectory}\` override. Please change it back to the default.`,
       };
     }
-    const versionRange = deps['next'];
-    if (!versionRange) {
+    const nextVersion = deps['next'];
```

Comment:
I think calling it `nextVersion` might be a bit confusing, since it's a range when it comes from `package.json`.

Might make sense to call it `nextVersionRange`.

Reply:
I see what you mean.

I think of a range as the `engines` field, not the `dependencies` field because a range can be `>= 12.1.3 <= 14.0.0`.

I left the todo comment so that we can come back to resolving the actual installed version using the lockfile (we won't be able to use require()/resolve() unfortunately since this code executes before `npm install`)